app/helpers/template_helper.rb

```diff
@@ -1,10 +1,13 @@
   module TemplateHelper
     def build_remediation_plan(remediation_ids, host)
-      entries = PreupgradeReportEntry.where(id: remediation_ids.split(','), host_id: host.id)
+      entries = PreupgradeReportEntry.where(id: remediation_ids, host: host)
+                    .to_a.map { |e| JSON.parse(e.detail) }
+                    .compact
```

Comment:
Since you're expecting there might be `nil` values, wouldn't it be better to filter them out on the database level? Not speaking about the fact that if there was a `nil` value somewhere, an exception would be raised before it would reach `compact`

Also instead of loading the entire record and then taking one field out of it, you could use `#pluck` with a symbol to get only the wanted field.